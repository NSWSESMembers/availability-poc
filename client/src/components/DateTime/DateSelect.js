import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import styles from './styles';

const DateSelect = ({
  date,
  select = false,
  edit = false,
  fade = false,
  onSelect,
  onEdit,
  availType = 'NotSpecified',
}) => {
  let dayStyles = [styles.day];
  let dayLabelStyles = [styles.dayLabel];
  const dayIconStyles = [styles.dayIcon];

  if (availType === 'OutOfRange') {
    dayStyles = [styles.dayOutOfRange];
    dayLabelStyles = [styles.dayLabelOutOfRange];
  }

  if (availType === 'Available') {
    dayStyles = [styles.dayAvailable];
    dayLabelStyles = [styles.dayLabelAvailable];
  }

  if (availType === 'Unavailable') {
    dayStyles = [styles.dayUnavailable];
    dayLabelStyles = [styles.dayLabelUnavailable];
  }

  if (availType === 'Urgent') {
    dayStyles = [styles.dayUrgent];
    dayLabelStyles = [styles.dayLabelUrgent];
  }

  if (select && availType === 'NotSpecified') {
    dayStyles = [styles.daySelect];
    dayLabelStyles = [styles.dayLabelSelect];
  }
  /*
  if (edit) {
    dayStyles = [styles.dayEdit];
    dayLabelStyles = [styles.dayLabelEdit];
  }
*/
  if (fade) {
    dayStyles.push({ backgroundColor: 'transparent', borderColor: '#AAA' });
    dayLabelStyles.push({ color: '#AAA' });
    dayIconStyles.push({ color: '#AAA' });
  }

  if (availType === 'OutOfRange') {
    return (
      <View style={dayStyles}>
        <Text style={dayLabelStyles}>{moment.unix(date).format('dd')}</Text>
        <Text style={dayLabelStyles}>{moment.unix(date).format('DD')}</Text>
        <Icon size={18} name="check-circle" color="white" />
      </View>
    );
  }

  if (availType === 'NotSpecified') {
    return (
      <TouchableOpacity style={dayStyles} onPress={() => onSelect(date)}>
        <Text style={dayLabelStyles}>{moment.unix(date).format('dd')}</Text>
        <Text style={dayLabelStyles}>{moment.unix(date).format('DD')}</Text>
        <Icon size={18} name="check-circle" color="white" />
      </TouchableOpacity>
    );
  }

  if (edit) {
    return (
      <TouchableOpacity style={dayStyles} onPress={() => onEdit(date)}>
        <Text style={dayLabelStyles}>{moment.unix(date).format('dd')}</Text>
        <Text style={dayLabelStyles}>{moment.unix(date).format('DD')}</Text>
        {availType === 'Available' && <Icon size={18} name="check-circle" color="white" />}
        {availType === 'Unavailable' && <Icon size={18} name="times-circle" color="white" />}
        {availType === 'Urgent' && <Icon size={18} name="exclamation-circle" color="white" />}
      </TouchableOpacity>
    );
  }

  return (
    <TouchableOpacity style={dayStyles} onPress={() => onEdit(date)}>
      <Text style={dayLabelStyles}>{moment.unix(date).format('dd')}</Text>
      <Text style={dayLabelStyles}>{moment.unix(date).format('DD')}</Text>
      {availType === 'Available' && <Icon size={18} name="check-circle" style={dayIconStyles} />}
      {availType === 'Unavailable' && <Icon size={18} name="times-circle" style={dayIconStyles} />}
      {availType === 'Urgent' && <Icon size={18} name="exclamation-circle" style={dayIconStyles} />}
    </TouchableOpacity>
  );
};

DateSelect.propTypes = {
  date: PropTypes.number.isRequired,
  select: PropTypes.bool,
  edit: PropTypes.bool,
  fade: PropTypes.bool,
  onSelect: PropTypes.func,
  onEdit: PropTypes.func,
  availType: PropTypes.string,
};

export default DateSelect;
