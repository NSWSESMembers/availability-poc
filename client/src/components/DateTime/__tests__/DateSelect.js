import React from 'react';
import renderer from 'react-test-renderer';

import DateSelect from '../DateSelect';

const noop = () => {};

test('renders correctly', () => {
  const tree = renderer.create(
    <DateSelect
      key="1"
      availType="OUT_OF_RANGE"
      day={1}
      onSelect={noop}
      select
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();

  const tree2 = renderer.create(
    <DateSelect
      key="1"
      availType="NotSpecified"
      day={1}
      onSelect={noop}
      select
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer.create(
    <DateSelect
      key="1"
      availType="AVAILABLE"
      day={1}
      onSelect={noop}
      select
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();

  const tree4 = renderer.create(
    <DateSelect
      key="1"
      availType="UNAVAILABLE"
      day={1}
      onSelect={noop}
      select
    />,
  ).toJSON();
  expect(tree4).toMatchSnapshot();

  const tree5 = renderer.create(
    <DateSelect
      key="1"
      availType="URGENT"
      day={1}
      onSelect={noop}
      select
    />,
  ).toJSON();
  expect(tree5).toMatchSnapshot();
});
