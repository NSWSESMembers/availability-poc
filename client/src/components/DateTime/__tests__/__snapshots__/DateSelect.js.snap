// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderColor": "#00a699",
      "borderWidth": 1,
      "flex": 1,
      "margin": 2,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Th
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    01
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderColor": "#00a699",
      "borderWidth": 1,
      "flex": 1,
      "margin": 2,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Th
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    01
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 18,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ÔÅò
  </Text>
</View>
`;

exports[`renders correctly 3`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderColor": "#00a699",
      "borderWidth": 1,
      "flex": 1,
      "margin": 2,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Th
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    01
  </Text>
</View>
`;

exports[`renders correctly 4`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderColor": "#00a699",
      "borderWidth": 1,
      "flex": 1,
      "margin": 2,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Th
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    01
  </Text>
</View>
`;

exports[`renders correctly 5`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderColor": "#00a699",
      "borderWidth": 1,
      "flex": 1,
      "margin": 2,
      "opacity": 1,
      "padding": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Th
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00a699",
          "fontWeight": "bold",
        },
      ]
    }
  >
    01
  </Text>
</View>
`;
